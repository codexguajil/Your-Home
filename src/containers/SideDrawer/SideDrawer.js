import React from 'react';
import { Drawer, Button } from 'antd';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

class SideDrawer extends React.Component {
  state = { 
    visible: false, 
    childrenDrawer: false, 
    placement: 'left'
  };

  showDrawer = () => {
    this.setState({
      visible: true,
    });
  };

  onClose = () => {
    this.setState({
      visible: false,
    });
  };

  renderProjects = () => {
    return this.props.projects.map(project => (
      <Link to={`/projects/${project.id}`} onClick={this.onClose} key={'link'+project.id}>
        <Button type="default" key={project.id} block>{project.name}</Button>
      </Link>
    ))
  }

  render() {
    return (
      <div id='menu'>
        <img src="https://img.icons8.com/ios/30/ffffff/menu-filled.png"
          onClick={this.showDrawer}
          alt='menu'>
        </img>
        <Drawer
          title="Your Projects"
          width={500}
          placement={this.state.placement}
          closable={false}
          onClose={this.onClose}
          visible={this.state.visible}
        >
          {this.renderProjects()}
          <div
            style={{
              position: 'absolute',
              bottom: 0,
              width: '100%',
              borderTop: '1px solid #e8e8e8',
              padding: '10px 16px',
              textAlign: 'right',
              left: 0,
              background: '#fff',
              borderRadius: '0 0 4px 4px',
            }}
          >
            <Button
              style={{
                marginRight: 8,
              }}
              onClick={this.onClose}
            >
              Cancel
            </Button>
          </div>
        </Drawer>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  projects: state.projects
});

export default connect(mapStateToProps)(SideDrawer);