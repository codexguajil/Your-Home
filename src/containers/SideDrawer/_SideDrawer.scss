#menu {
  display: inline;
  position: absolute;
  left: 25px;
  top: 25px;
}

.side-drawer-btn {
  margin-bottom: 15px;
  height: 50px;
}