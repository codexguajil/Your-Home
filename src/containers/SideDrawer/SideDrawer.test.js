import { SideDrawer } from './SideDrawer';
import React from 'react';
import { shallow } from 'enzyme';

describe('SideDrawer', () => {
  let wrapper;
  let mockProjects

  beforeEach(() => {
    mockProjects = ['1', '2']
    wrapper = shallow(<SideDrawer projects={mockProjects}/>)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should setState on showDrawer', () => {
    wrapper.instance().showDrawer();
    expect(wrapper.state('visible')).toBe(true);
  });

  it('should setState on onClose', () => {
    wrapper.instance().onClose();
    expect(wrapper.state('visible')).toBe(false);
  });

});