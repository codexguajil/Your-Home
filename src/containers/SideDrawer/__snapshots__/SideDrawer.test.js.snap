// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideDrawer should match snapshot 1`] = `
<div
  id="menu"
>
  <img
    alt="menu"
    onClick={[Function]}
    src="https://img.icons8.com/ios/30/ffffff/menu-filled.png"
  />
  <withConfigConsumer(Drawer)
    closable={false}
    onClose={[Function]}
    placement="left"
    title="Your Projects"
    visible={false}
    width={500}
  >
    <Link
      key="linkundefined"
      onClick={[Function]}
      to="/projects/undefined"
    >
      <Button
        block={true}
        className="side-drawer-btn"
        ghost={false}
        htmlType="button"
        loading={false}
        type="default"
      />
    </Link>
    <Link
      key="linkundefined"
      onClick={[Function]}
      to="/projects/undefined"
    >
      <Button
        block={true}
        className="side-drawer-btn"
        ghost={false}
        htmlType="button"
        loading={false}
        type="default"
      />
    </Link>
    <div
      style={
        Object {
          "background": "#fff",
          "borderRadius": "0 0 4px 4px",
          "borderTop": "1px solid #e8e8e8",
          "bottom": 0,
          "left": 0,
          "padding": "10px 16px",
          "position": "absolute",
          "textAlign": "right",
          "width": "100%",
        }
      }
    >
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
        style={
          Object {
            "marginRight": 8,
          }
        }
      >
        Cancel
      </Button>
    </div>
  </withConfigConsumer(Drawer)>
</div>
`;
